frontend/apps/reader/modules/readerstatus.lua

```diff
@@ -36,19 +37,116 @@ function ReaderStatus:addToMainMenu(menu_items)
+                    enabled = collate,
+                    callback = function()
+                        self:openNextFile(self.document.file)
+                        UIManager:close(choose_action)
+                    end,
+                },
+                {
+                    text = _("File Browser"),
+                    callback = function()
+                        self:openFileBrowser()
+                        UIManager:close(choose_action)
+                    end,
+                },
+            },
+        }
+        choose_action = ButtonDialogTitle:new{
+            title = _("You've reached the end of the document.\nWhat would you like to do?"),
+            title_align = "center",
+            buttons = buttons,
+        }
+
+        UIManager:show(choose_action)
+    elseif settings == "book_status" then
         self:showStatus()
+    elseif settings == "next_file" then
+        if G_reader_settings:readSetting("collate") ~= "access" then
+            local info = InfoMessage:new{
+                text = _("Searching next file…"),
+            }
+            UIManager:show(info)
+            UIManager:forceRePaint()
+            self:openNextFile(self.document.file)
+            UIManager:close(info)
+        else
+            UIManager:show(InfoMessage:new{
+                text = _("Could not open next file. Sort by last read date does not support this feature."),
+            })
+        end
+    elseif settings == "file_browser" then
+        self:openFileBrowser()
+    elseif settings == "book_status_file_browser" then
+        local on_exit_func = function() self:openFileBrowser() end
+        self:showStatus(on_exit_func)
     end
 end
 
-function ReaderStatus:showStatus()
+function ReaderStatus:openFileBrowser()
+    local FileManager = require("apps/filemanager/filemanager")
+    if not FileManager.instance then
+        self.ui:showFileManager()
+    end
+    self.ui:onClose()
+    self.document = nil
+end
+
+function ReaderStatus:openNextFile(next_file)
+    local FileManager = require("apps/filemanager/filemanager")
+    if not FileManager.instance then
+        self.ui:showFileManager()
+    end
+    next_file = FileManager.instance.file_chooser:getNextFile(next_file)
+    FileManager.instance:onClose()
+    local ReaderUI = require("apps/reader/readerui")
+    if next_file then
+        ReaderUI:showReader(next_file)
+    else
+        UIManager:show(InfoMessage:new{
+            text = _("This is the last file in the current folder. No next file to open."),
+        })
+    end
+end
+
+function ReaderStatus:showStatus(on_exit_func)
     local status_page = BookStatusWidget:new {
         thumbnail = self.document:getCoverPageImage(),
         props = self.document:getProps(),
         document = self.document,
         settings = self.settings,
         view = self.view,
     }
+    if on_exit_func then
+        on_exit_func()
+    end
     UIManager:show(status_page)
```

Comment:
We cannot do that :)
In that case when we run `self:showStatus` after `self:openFileBrowser()` we get 
`attempt to index field 'document' (a nil value)` because self.document is nil after run `openFileBrowser`

Reply:
OK, I see, so now, it's no more a `on_exit_func`, but a `before_show_callback` :)
(It also looks less natural than it did before)